lib/xandra/protocol.ex

```diff
@@ -309,8 +311,13 @@ defmodule Xandra.Protocol do
     <<value::64>>
   end
 
-  defp encode_value(:date, value) when value in 0..0xFFFFFFFF do
-    <<value::32>>
+  defp encode_value(:date, %Date{} = value) do
+    value = Date.diff(value, ~D[1970-01-01])
+    <<(value + @unix_epoch_days)::32>>
+  end
+
+  defp encode_value(:date, value) when value in -0x80000000..0x7FFFFFFF do
```

Comment:
Looks like we can use `-@unix_epoch_days` in the guard here?

Reply:
Maybe we can also give a name to `0x7FFFFFFF` so it doesn't look like a magic number.

Reply:
Not sure `0x7FFFFFFF` can have any name here. 
Though that range could be written as `-@unix_epoch_days..(@unix_epoch_days - 1)`.
Does this read better?

Reply:
That reads way better yes.